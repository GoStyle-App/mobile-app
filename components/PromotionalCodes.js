import React from 'react';
import { StyleSheet, View, Text, Button, FlatList } from 'react-native';
import PromotionalCode from './PromotionalCode';

export default function PromotionalCodes({ codes }) {
    let show = false;

    console.log(codes, 'banane');
    console.log(typeof codes);
    console.log(codes.length);
    codes.forEach(code => {
        console.log(code.unlock);
        if (code.unlock) {
            show = true;
        }
    });

    console.log(show);

    if (show) {
        return (
            <View style={ styles.codes }>
                <Text style={ styles.title }>Mes codes promos</Text>
                <FlatList
                    data={ codes }
                    keyExtractor={(item) => item.id.toString()}
                    renderItem={({ item }) => <PromotionalCode code={ item } />}
                />
            </View>
        )
    }

    return (
        <View style={ styles.codes }>
            <Text style={ styles.title }>Mes codes promos</Text>
            <Text style={ styles.text }>Pas de codes promos</Text>
            <View style={ styles.buttonContainer }>
                <Button
                    title="Scanner"
                    onPress={() => {
                        //Todo navigate scan page
                    }}
                    color='#38B6FF'
                />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    codes: {
        marginTop: 80,
        alignItems: 'center',
    },
    title: {
        marginBottom: 30,
        color: '#2E2E2E',
        fontSize: 20,
        textTransform: 'uppercase',
    },
    text: {
        marginBottom: 30,
        color: '#737373',
    },
    buttonContainer: {
        width: 150,
    }
})
