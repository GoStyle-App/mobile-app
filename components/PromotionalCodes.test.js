import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import PromotionalCodes from './PromotionalCodes';
import Scan from './Scan';
import 'isomorphic-fetch';

describe('Testing PromotionalCodes Component', () => {
    const myCodes = [];
    const route = {
        'key': 'Home-Nx8Rj_yGS8oGfmw-QfVCq',
        'name': 'Home',
        'params': {
            'link': ''
        },
    };
    const navigation = {
        'addListener': function addListener(){},
        'canGoBack': function canGoBack(){},
        'closeDrawer': function anonymous(){},
        'dangerouslyGetParent': function dangerouslyGetParent(){},
        'dangerouslyGetState': function anonymous(){},
        'dispatch': function dispatch(){},
        'goBack': function anonymous(){},
        'isFocused': function isFocused(){},
        'jumpTo': function anonymous(){},
        'navigate': function anonymous(){},
        'openDrawer': function anonymous(){},
        'removeListener': function removeListener(){},
        'reset': function anonymous(){},
        'setOptions': function setOptions(){},
        'setParams': function anonymous(){},
        'toggleDrawer': function anonymous(){},
    };

    test('The app prints the corresponding screen', async () => {
        const component = (
            <NavigationContainer>
                <PromotionalCodes route={ route } navigation={ navigation } myCodes={ myCodes } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const titleScreen = await findByText('Mes codes promos');

        expect(titleScreen).toBeTruthy();
    });

    test('Page contains the Scan button', async () => {
        const component = (
            <NavigationContainer>
                <PromotionalCodes route={ route } navigation={ navigation } myCodes={ myCodes } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const buttonToScan = await findByText('Scan');

        expect(buttonToScan).toBeTruthy();
    });

    test('Scan Button prints Scan Screen on press', async () => {
        const component = (
            <NavigationContainer>
                <PromotionalCodes route={ route } navigation={ navigation } myCodes={ myCodes } />
            </NavigationContainer>
        );
        const { findByText } = render(component);
        const oldScreen = await findByText('Mes codes promos');
        const buttonScan = await findByText('Scan');

        expect(oldScreen).toBeTruthy();
        expect(buttonScan).toBeTruthy();

        fireEvent(buttonScan, 'press');

        const newScreen = render(<Scan />).toJSON();

        expect(newScreen).toMatchSnapshot();
    });
});
