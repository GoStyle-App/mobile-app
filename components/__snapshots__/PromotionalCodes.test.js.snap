// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PromotionalCodes Component Scan Button prints Scan Screen on press 1`] = `
<Text>
  Requesting for camera permission
</Text>
`;
